### ISSUE TITLE
x is defined but never used

### ISSUE THREAD

**Issue by epegzz** [babel/babel-eslint#179]
That error should actually not be thrown for this code, right?

```jsx

let x;
_.each([1, 2], n => {
  x = n;
});

```

**Comment by hzoo** [comment]
can you reproduce with the regular parser?

**Comment by epegzz** [comment]
okay, actually eslint complains as well when i run this code:

```js
var x;
_.each([1, 2], function (n) {
  x = n;
});
```

**Comment by epegzz** [comment]
Okay, found the explanation here: https://github.com/eslint/eslint/issues/3417

```This rule considers a variable as used when the variable value is either assigned to some other variable or used in the expression. Just assigning values to a variable doesn't mean that it's used somewhere.```
